### PR TITLE
docs: add method docs

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

I have been researching EIP-2678 and thought that this package should be updated.
Most of the material is borrowed from the EIP as well as fitting in existing documentation.

### How I did it

Read through https://eips.ethereum.org/EIPS/eip-2678#package-name and add bits to PackageManifest and ContractType classes.

### How to verify it

Verify the docs make sense that were added

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: ethpm_types/contract_type.py [hunk 1]
```diff
@@ -187,14 +260,31 @@ class ContractType(BaseModel):
...
     runtime_bytecode: Optional[Bytecode] = Field(None, alias="runtimeBytecode")
-    # abi, userdoc and devdoc must conform to spec
+    """The unlinked 0x-prefixed runtime portion of bytecode for this ContractType."""
+
     abi: List[ABI] = []
+    """The application binary interface to the contract."""
+
     sourcemap: Optional[SourceMap] = None  # NOTE: Not a part of canonical EIP-2678 spec
+    """
+    **NOTE**: This is not part of the canonical EIP-2678 spec
```
Comment (fubuloubu) [inline ethpm_types/contract_type.py:285]: This is noted above, I would say the docstrings here should have some substance, and either keep the comment note or move it into the docstring

File: ethpm_types/manifest.py [hunk 2]
```diff
@@ -46,33 +50,96 @@ def __repr__(self):
...
     version: Optional[str] = None
-    # NOTE: ``meta`` should be in all published packages
+    """The version of the release, which should be SemVer."""
+
     meta: Optional[PackageMeta] = None
-    # NOTE: ``sources`` source tree should be necessary and sufficient to compile
-    #       all ``ContractType``s in manifest
+    """
+    Important data that is not integral to installation
+    but should be included when publishing.
```
Comment (fubuloubu) [inline ethpm_types/manifest.py:102]: The note was removed from above, but add the recommendation that all published packages should have the meta field

File: ethpm_types/manifest.py [hunk 3]
```diff
@@ -46,33 +50,96 @@ def __repr__(self):
...
+    :class:`~ethpm_types.contract_type.ContractType`s and other deployment details
+    for those deployed contract instances.
+    The set of chains defined by the BIP122 URI keys for this object must be unique.
+    There cannot be two different URI keys in a deployments field representing the same blockchain.
+    """
+
     dependencies: Optional[Dict[PackageName, AnyUrl]] = Field(None, alias="buildDependencies")
+    """
+    A mapping of EthPM packages that this project depends on.
+    The values must be content-addressable URIs that conformsto the same
```
Comment (fubuloubu) [inline ethpm_types/manifest.py:140]:
```suggestion
    The values must be content-addressable URIs that conforms to the same
```

**Review by fubuloubu — APPROVED** [review]
Do we want to add the documentation generator for this package so that `docs.apeworx.io/ethpm-types` gets published?

File: ethpm_types/manifest.py (hunk 2 above)
Comment (unparalleled-js) [reply ethpm_types/manifest.py:102]: Ah, I tried when by saying `but should be included when publishing` but I will adjust it

**Review by NotPeopling2day — APPROVED** [review]

**Pull request merged**